test/vfd.c

```diff
@@ -4352,6 +4352,984 @@ test_vector_io(const char *vfd_name)
     return -1;
 } /* end test_vector_io() */
 
+/*-------------------------------------------------------------------------
+ * Function:    test_selection_io_write
+ *
+ * Purpose:     Updates write buffers to ensure a unique value is written
+ *              to each element and issues a selection write call.
+ *
+ * Return:      Success:        TRUE
+ *              Failure:        FALSE
+ *
+ * Programmer:  Neil Fortner
+ *              7/1/21
+ *
+ * Changes:     None.
+ *
+ *-------------------------------------------------------------------------
+ */
+/* Array dimensions, used for all selection I/O tests.  Currently both must be
+ * even.  1-Dimensional arrays have a size of SEL_IO_DIM0 * SEL_IO_DIM1. */
+#define SEL_IO_DIM0 8
+#define SEL_IO_DIM1 10
+
+static hbool_t
+test_selection_io_write(H5FD_t *lf, H5FD_mem_t type, uint32_t count, hid_t mem_spaces[], hid_t file_spaces[],
+                        haddr_t offsets[], size_t element_sizes[], int *wbufs[])
+{
+    int     i;
+    int     j;
+    hbool_t result = TRUE;
+
+    /* Update write buffer */
+    for (i = 0; i < (int)count; i++)
+        if (wbufs[i] && (i == 0 || wbufs[i] != wbufs[i - 1]))
+            for (j = 0; j < SEL_IO_DIM0 * SEL_IO_DIM1; j++)
+                wbufs[i][j] += 2 * SEL_IO_DIM0 * SEL_IO_DIM1;
+
```

Comment:
Am I missing something, or are you always setting wbufs[I][j] = 160 here?

Reply:
Every time we run a new test case we *increment* by 160 so that each value we write is unique and if something goes wrong and we get data from a previous write we'll know its wrong.